packages/kernel/src/worker-runtime.test.ts

```diff
@@ -215,7 +227,222 @@ assert len(w) == 0, f"Warning occurred: {w[0].message if w else None}"
+      );
+      expect(firstItem.textEdit.range.end).toEqual(
+        expect.objectContaining({ line: 2, character: 5 })
+      );
+    });
+
+    test("should return suggestions for a module when no prefix after the cursor is present", async () => {
+      // Should give suggestions for a module when no prefix is present
+      const code = `import streamlit as st
+st.
+`;
+      const autocompleteResults = await getCodeCompletions(
+        {
+          code: code,
+          currentLine: "st.",
+          currentLineNumber: 2,
+          offset: 3,
+        },
+        pyodide as PyodideInterface
+      );
+
+      expect(
+        autocompleteResults.items
+          .map((item: { label: string }) => item.label)
+          .slice(0, 8)
+      ).toEqual([
+        "altair_chart",
+        "area_chart",
+        "audio",
+        "audio_input",
+        "balloons",
+        "bar_chart",
+        "bokeh_chart",
+        "button",
+      ]);
+    });
+
+    test("should return and prioritize function arguments", async () => {
+      // Should give function arguments suggestions
+      const code = `import streamlit as st
+x = {}
+st.title()
+`;
+      const autocompleteResults = await getCodeCompletions(
+        {
+          code: code,
+          currentLine: "st.title()",
+          currentLineNumber: 3,
+          offset: 9,
+        },
+        pyodide as PyodideInterface
+      );
+
+      // the code editors use sortText to sort the items in the list
+      // before showing them on the UI
+      // function arguments have always bigger priority then other suggestions
+      expect(
+        autocompleteResults.items
+          .map((item: { sortText: string }) => item.sortText)
+          .sort()
+          .slice(0, 3)
+      ).toEqual(["aaanchor=", "aabody=", "aahelp="]);
+    });
+
+    test("should give suggestions for local functions", async () => {
+      // Should give suggestions for local functions
+      const code = `import json
+def handle(param_1: int, limit: str = "default") -> str:
+  """
+  This function returns the parameters as a string.
+  """
+  return f"Result - param_1: {param_1}, limit: {limit}"
+
+hand
+`;
+      const autocompleteResults = await getCodeCompletions(
+        {
+          code: code,
+          currentLine: "handle",
+          currentLineNumber: 8,
+          offset: 4,
+        },
+        pyodide as PyodideInterface
+      );
+
+      expect(
+        autocompleteResults.items.map(
+          (item: { label: string; documentation: string }) => ({
+            label: item.label,
+            documentation: item.documentation.trim(),
+          })
+        )
+      ).toEqual([
+        {
+          label: "handle",
+          documentation: "This function returns the parameters as a string.",
+        },
+      ]);
+    });
+
+    test("should handle invalid requests and return empty response", async () => {
+      const code = `import math
+      math.cos()
```

Comment:
Are these indentation correct as intended?

Reply:
The `jedi` parser didn't complained, seems like it is less strict and maybe it makes sense. 
In this case it is working because I am asking for suggestions for line 3 and it doesn't exist and we are getting empty results back. Regardless, both cases are handled.